### DISCUSSION TITLE
OOMKilled Brokers since Java17 Update

### DISCUSSION THREAD

**Discussion by siegenthalerroger** [strimzi/strimzi-kafka-operator#8241] [Q&A] [answered]
Hi,

We've identified an issue with our strimzi powered kafka cluster that is causing regular (ca 5. day intervals) OOMKilled restarts of our brokers. We've been able to locate the issue as being caused by our upgrade from Strimzi 0.32.0 to 0.33.1 and/or the upgrade from Kafka 3.3.1 to 3.3.2. We are also still observing this behaviour on the latest strimzi 0.33.2 and kafka 3.4.0. 

The images below show a distinct change in GC behaviour and an analysis of the changes in the updates leads us to believe that the java17 update has led to the encountered issue. I am wondering if there's any specific GC properties that could be set to tame the new behaviour or if this is potentially a memory leak on the kafka side (see last image).

**Before (strimzi 0.32.0, kafka 3.3.1, java 11):**
<img width="1711" alt="image" src="https://user-images.githubusercontent.com/6848427/225650245-ec00cd6a-7002-4c65-a2c6-b6af5d22b177.png">

**After (strimzi 0.33.1, kafka 3.3.2, java 17):**
<img width="1713" alt="image" src="https://user-images.githubusercontent.com/6848427/225650193-c550bdea-6194-4d99-8321-cd5710f3bfcb.png">

**Memory Leak?** Note the run-up of memory with GC unable to bring it back down (apparently)
<img width="1696" alt="image" src="https://user-images.githubusercontent.com/6848427/225673784-6981ee8c-d707-4560-b9eb-c1aab7ff1af2.png">

**Comment by scholzj** [comment, marked as answer]
Is the container killed by the OOM killer? I think in that case, maybe you should tune the resources of the container / Java memory settings as that suggests that Java tries to consume more memory than it should. You can also use the [`jvmOptions` section](https://strimzi.io/docs/operators/latest/full/configuring.html#con-common-configuration-jvm-reference) to tune the various JVM settings if you want.

**Reply by siegenthalerroger** [reply]
Yeah it was. Thanks for the tip about the options, just saw that they were misconfigured (not enough free space at all). I am left wondering about the endless growth of the jvm memory, but I'll keep an eye on it with the much lowered heap maxima.

**Reply by scholzj** [reply]
The GC behavior changes are interesting as well. But to be honest, my Java knowledge is not deep enough to know the details of what changed.
